### ISSUE TITLE
Docs say 'subpixel' instead of 'subpixels' for  method keyword in aperture_photometry

### ISSUE THREAD

**Issue by havijw** [astropy/photutils#1812]
Currently, [documentation for `aperture_photometry`](https://photutils.readthedocs.io/en/stable/api/photutils.aperture.aperture_photometry.html#photutils.aperture.aperture_photometry) gives `'subpixel'` as an option for the `method` keyword, when the correct option is `'subpixels'`. This also applies to the `do_photometry` methods for each `Aperture` subclass; see for example [`CircularAperture`](https://photutils.readthedocs.io/en/stable/api/photutils.aperture.CircularAperture.html#photutils.aperture.CircularAperture.do_photometry).

**Comment by larrybradley** [comment]
@havijw `subpixel` is the correct option for the `method` keyword.  I think you've confused that with the separate `subpixels` keyword.  If `method='subpixel'`, then the `subpixels` keyword defines the number of subpixels.  So, a correct call to `aperture_photometry` would be:

```
>>> aperture_photometry(data, aperture, method='subpixel', subpixels=5)
```

**Comment by havijw** [comment]
@larrybradley thank you for the correction. I had a problem while working with regions since the keyword there is `subpixels` and misidentified the source. I think there's still a problem, albeit a more minor one, where the check for a valid `subpixels` argument will never run because it's checking if `mode == 'subpixels'`:
```python
if mode == 'subpixels':
    if not isinstance(subpixels, int) or subpixels <= 0:
        raise ValueError('subpixels must be a strictly positive '
                         'integer')`
```
See <https://github.com/astropy/photutils/blob/main/photutils/aperture/core.py#L193-L196>.

**Comment by larrybradley** [comment]
@havijw Ack!  Yes, that's a bug.  Fixed in #1816.  Thanks!
